### PR TITLE
dapp: fix dapp test for use with libraries

### PR DESCRIPTION
**Pull request by MrChico**
Addresses #426, @realisation please try it out

### CODE REVIEW

**Comment by MrChico** [comment]
For this to work you need to run dapp test with the environment var `DAPP_BUILD_EXRACT` set to 1

File: src/dapp/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Fixed
+Support for libraries. To test a project which includes libraries you need to use:
```
Comment (asymmetric) [inline src/dapp/CHANGELOG.md:10]:
```suggestion
- Support for libraries. To test a project which includes libraries you need to use:
```

Or something like that :)

File: src/dapp/libexec/dapp/dapp---make-library-state [hunk 2]
```diff
@@ -18,20 +18,11 @@ if [ "$DAPP_VERBOSE" ]; then set -x; fi
 # history tracking.)
 
 sender=0x00a329c0648769a73afac7f9381e08fb43dbea72
-nonce=1048576
+nonce=48576
```
Comment (rainbreak) [inline src/dapp/libexec/dapp/dapp---make-library-state:21]: Do you know what caused this to change? The previous value was valid at some point.

Comment (MrChico) [reply src/dapp/libexec/dapp/dapp---make-library-state:21]: afaik, the value is not relevant, it just needs to match whats in dapp build. `dapp address` couldn't handle such a high nonce value though.

**Pull request merged**